mws/apis/products.py

```diff
@@ -163,9 +163,89 @@ def get_lowest_priced_offers_for_asin(
         }
         return self.make_request(data)
 
-    # # # TODO add this
-    # def get_my_fees_estimate(self):
-    #     pass
+    def get_my_fees_estimate(
+        self, marketplace_id, id_type, id_value, is_amazon_fulfilled, identifier, listing_price, listing_price_currency_code
+    ):
+
+        """
+        Each argument must be a list, up to a length of 20, with id_value (number of ASINs or SellerSKUs) determining how many requests are made.
+        Options: 
+            A single element list which repeats, or
+            A list of equal length as id_value for a one to one relationship
+
+        Example:
+            marketplace_id = ['A1F83G8C2ARO7P']
+            id_value = ['B082YTKC47', 'B07WGJ9JGP', 'B01LXP5TXI']
+            id_type = ['ASIN']
+            is_amazon_fulfilled = [True, False, True]
+            identifier = ['request1', 'request2', 'request3']
+            listing_price = [9, 16.42, 2.99]
+            listing_price_currency_code = ['GBP']
+
+        Example 2:
+            marketplace_id = ['A1F83G8C2ARO7P', 'ATVPDKIKX0DER', 'A13V1IB3VIYZZH']
+            id_value = ['B082YTKC47', 'B07WGJ9JGP', 'B01-LXP-TXI6']
+            id_type = ['ASIN', 'ASIN', 'SellerSKU']
+            is_amazon_fulfilled = [True]
+            identifier = ['Fee requests']
+            listing_price = [19]
+            listing_price_currency_code = ['GBP', 'USD', 'EUR']
+            
+            
+        Returns Amazon Fees for ASINs and SellerSKUs specified.
+
+        Docs:
+        https://docs.developer.amazonservices.com/en_UK/products/Products_GetMyFeesEstimate.html
+
+        """
+        
+        data = {
+            "Action": "GetMyFeesEstimate",
+        }
+
+        values = []
+        for i in range(len(id_value)):
+            values_dict = dict()
+
+            try:
+                values_dict['MarketplaceId'] = marketplace_id[i]
+            except IndexError:
+                values_dict['MarketplaceId'] = marketplace_id[0]
```

Comment:
I can see how this is attempting to use a list of values that contains either 1 or *n* items, where *n* is intended to be the same as the number of `id_value` elements. However, in the case that any given list is shorter than it should be, this will simply return the first element for each subsequent `id_value`:

```python
id_value = [0, 1, 2, 3]
marketplace_id = ['a', 'b']
```

The above would create sets like `(0, 'a'), (1, 'b'), (2, 'a'), (3, 'a')` (the last two re-using the first element in `marketplace_id`). This may create an error if the data in position 0 is not meant to match up with the data in position *x* of any other list. Trying to account for this error means checking list length for every list before processing.

That point is moot, however, as the design becomes too cumbersome to manage with these tightly-coupled list arguments (*see other comments on review*).